### PR TITLE
Linking in prerequisite sections

### PR DESCRIPTION
**Pull request by rudri182**
Linked sections in prerequisite sections. issue #731

### CODE REVIEW

**Comment by rudri182** [comment]
I have resolved conflicts that I understood. Suggest changes that are required. 😊

**Comment by rudri182** [comment]
I have also linked sections of SAAS and PYTHON. I am still working on this to link prerequisites in some sections where there is none or not available.

**Comment by rudri182** [comment]
> @alodahl I found that there is JS lessons 5 too...so it could be added here?

Yes, there is a total of 9 lessons in javascript. Can we add all these to here? Because recursion is the concept that can be used anywhere. So if all lessons of JS are linked it is easy to refer. 🙂 This is what I feel. If it is appropriate to add.

**Comment by Kundan28** [comment]
> Yes, there is a total of 9 lessons in javascript. Can we add all these to here? Because recursion is the concept that can be used anywhere. So if all lessons of JS are linked it is easy to refer. 🙂 This is what I feel. If it is appropriate to add.

Let's wait for @alodahl reply.

**Comment by alodahl** [comment]
The point of the prerequisites is to be sure the learner has enough background to understand the new lesson - not to list every lesson that came before it. For example, you don’t need to understand object oriented programming to use recursion. 

If you think there is something that they shouldn’t miss before learning recursion, add it; otherwise leave it as is. @rudri182 @Kundan28

File: agile-development/scrum.md [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
 ### Prerequisites
-HTML  
-CSS  
-Git  
-GitHub  
-Editor  
-Desire to learn  
+- [HTML](https://github.com/Techtonica/curriculum/blob/master/html/html.md)  
+- [CSS](https://github.com/Techtonica/curriculum/blob/master/css/css.md)
+- [Git](https://github.com/Techtonica/curriculum/tree/master/git-version-control)
```
Comment (alodahl) [inline agile-development/scrum.md]: Link specific lesson, not a folder please.

File: express-js/express.md [hunk 2]
```diff
@@ -4,8 +4,8 @@
 4 hours, including lab exercise
 
 ### Prerequisites
-- How the Internet Works
-- Node.js
+- [How the Internet Works](https://github.com/Techtonica/curriculum/blob/master/requests-and-responses/requests-and-responses.md)
+- [Node.js](https://github.com/Techtonica/curriculum/tree/master/node-js)
```
Comment (alodahl) [inline express-js/express.md]: Link specific file

File: recursion/recursion.md [hunk 3]
```diff
@@ -7,11 +7,11 @@ About 4.5 hours
...
-- JS 1 lesson
-- JS 2 lesson
-- JS 3 lesson
-- JS 4 lesson
-- Score of __15__ or higher at Coding Challege Assesment, Level 1
+- [JS 1 lesson](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md)
+- [JS 2 lesson](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-2.md)
+- [JS 3 lesson](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-3.md)
+- [JS 4 lesson](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-4.md)
+- Score of __15__ or higher at Coding Challenge Assessment, Level 1
```
Comment (alodahl) [inline recursion/recursion.md]: Please remove this line... the assessments no longer have a points system

File: projects/eventonica-project.md [hunk 4]
```diff
@@ -164,16 +199,29 @@ Users and a Events table.
...
+
 Once you feel comfortable about understanding what the code is doing, commenting out the example search code and implement your custom search code in `app.searchEventful`.
 
+
 Use Inquier to:
 1. Ask app user to input a keyword to search on Eventful, example: "dancing", "fashion", etc.
 2. Display the first result that comes back from Eventful, and ask app user if he or she would like to save this to the Postgres database.
 3. If app user chooses yes, then make that save.
-4. If app user chooses no, then go back to step 1 of part 4 
+4. If app user chooses no, then go back to step 1 of part 4
```
Comment (alodahl) [inline projects/eventonica-project.md]: Why is this edit here?

File: projects/eventonica-project.md [hunk 5]
```diff
@@ -100,20 +115,34 @@ We'll get to that in a future week.
...
 
 **Note:** In this project, the instructions to complete the project after the
 installation & setup steps will be increasingly vague. This is done
 intentionally to get you used to thinking like a software engineer.
 
 #### Part 1 - Installation and Setup
 
+
+1. Make sure Sequel Pro is installed properly by creating a test table and query all of its entries
+2. Each group forks the starter code, and then every apprentice clones the forked version to her computer
```
Comment (alodahl) [inline projects/eventonica-project.md]: Why is this here?

File: projects/eventonica-project.md (hunk 4 above)
Comment (rudri182) [reply projects/eventonica-project.md]: Maybe it is because some files were conflicting when I pushed it. So I resolved conflicts. There was a duplication of some lines, therefore, I deleted one line and kept one line as it is. 😊

**Comment by rudri182** [comment]
I have modified the files in which changes were requested. 🎨

**Comment by rudri182** [comment]
@alodahl may I need to link separate files for **JavaScript Basics** in a prerequisite section or I can link folder in it?

**Comment by alodahl** [comment]
@rudri182 if the Javascript folder is already linked, you can leave it that way.

**Comment by rudri182** [comment]
> @rudri182 if the Javascript folder is already linked, you can leave it that way.

Sorry for the delaying reply. 
But what if it is not inked? I have linked the folder instead of lessons separately.

**Comment by alodahl** [comment]
@rudri182 please send me a link and line number of what you’re referring to.

**Comment by rudri182** [comment]
https://github.com/Techtonica/curriculum/pull/740/commits/645ffb0954a6553515853c7ecce4b8789151dc51

I have separated the lessons in deploying.md

File: projects/eventonica-project.md (hunk 5 above)
Comment (rudri182) [reply projects/eventonica-project.md]: > But why are these files here? The issue is for prereq sections.

I linked sections in **Languages** section. I have not modified other lines. I thought it would be easy for someone who is learning this lesson. 😊

Comment (alodahl) [reply projects/eventonica-project.md]: That’s great, but there’s a lot of new tutorial code here, too. Please fix this branch or make a new one with only changes for this issue. Google it or ask on the slack git channel if you’re not sure how.

File: projects/eventonica-project.md (hunk 4 above)
Comment (alodahl) [reply projects/eventonica-project.md]: Unless the changes you’re resolving are yours and new, best practice is to keep the master version in a merge conflict. I only learned that recently.

**Comment by rudri182** [comment]
Yes I will 😊. I read it today that it is better to keep original text 🙂 I
will do it as soon as possible.

**Comment by rudri182** [comment]
```
Part 1 - Installation and Setup
Fork the starter code to your own GitHub account, then clone your fork to your local machine. Don't forget to cd into your new folder =)

Create a blank keys.js file in your project root.

Go to Eventful's API page, create an account, and apply for your own API key. The application process for an API key is instant.

Tell your app about your new Eventful API key as well as your PostgreSQL database by adding the following to your keys.js file:

module.exports = {
  "eventful": '<YOUR_OWN_EVENTFUL_API_KEY_HERE>',
  "dbPassword": '<YOUR_EVENTONICA_POSTGRES_PASSWORD_HERE>'
}
Now uncomment the dbConnect() line. You should see the console produce some lines about Eventonica, YAY!

PAUSE. Help your group members finish Part 1.
```

May I change the file like this? Is it appropriate to copy and paste the lines from it?
I referred it from [eventonica-project](https://github.com/Techtonica/curriculum/blob/master/projects/eventonica-project.md). Is it up-to-date with a master branch?

**Comment by rudri182** [comment]
When I try to edit machine learning.md it shows the file which is already modified. So what should I do? It is showing [editing machine learning](https://github.com/rudri182/curriculum/edit/Linking-in-prerequisite-sections/roles-in-tech/machine-learning.md?pr=%2FTechtonica%2Fcurriculum%2Fpull%2F740). 
But in commits, it is showing an old file. So I am confused. How to correct it?

File: roles-in-tech/machine-learning.md [hunk 6]
```diff
@@ -36,10 +39,35 @@
 
 ### Materials
 
+- [Example video (10 min)](https://example.com)
+- [Other example article(20 min read)](https://otherexample.com)
```
Comment (saberprashant) [inline roles-in-tech/machine-learning.md]: Hi @rudri182 , please remove this link (https://otherexample.com), as it's an appropriate link. It will create a bad impact on users who'll click it(actually, it has some inappropriate content).

File: deploying/deploying.md [hunk 7]
```diff
@@ -4,19 +4,22 @@
...
+- [Express](https://github.com/Techtonica/curriculum/blob/master/express-js/express.md)
+- [MongoDB](https://github.com/Techtonica/curriculum/blob/master/databases/mongo-db.md)
+
 
 ### Motivation
 Deployment is a fancy term for getting your website on the web. After building out your app, you might want to share it with others, and to do that you need to deploy your app to the web where others can access it. In this lesson we'll learn more about deployment, and learn one way to deploy an app.
 
 ### Objectives
-**Apprentices will be able to:** 
+**Apprentices will be able to:**
```
[the remark this answers is not in the document]
Comment (rudri182) [reply deploying/deploying.md]: Yes, sure I will. Thanks for noticing. 🙂

File: agile-development/scrum.md [hunk 8]
```diff
@@ -5,12 +5,12 @@
...
-Git  
-GitHub  
-Editor  
-Desire to learn  
+- [HTML](https://github.com/Techtonica/curriculum/blob/master/html/html.md)  
+- [CSS](https://github.com/Techtonica/curriculum/blob/master/css/css.md)
+- [Git](https://github.com/Techtonica/curriculum/blob/master/git-version-control/git-version-control.md)  
+- [GitHub](https://github.com/Techtonica/curriculum/blob/master/github-code-storage/github-storage.md)  
+- Editor  
+- Desire to learn
```
[the remark this answers is not in the document]
Comment (rudri182) [reply agile-development/scrum.md]: Thank You 😊

File: roles-in-tech/machine-learning.md [hunk 9]
```diff
@@ -36,10 +39,35 @@
 
 ### Materials
 
+- [Example video (10 min)](https://example.com)
```
Comment (rudri182) [inline roles-in-tech/machine-learning.md]: I think this link should also be removed. Is it proper to remove? @saberprashant @alodahl

File: deploying/deploying.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (rudri182) [reply deploying/deploying.md]: Is it looking as good as `master` now? @saberprashant

File: roles-in-tech/machine-learning.md (hunk 9 above)
Comment (Kundan28) [reply roles-in-tech/machine-learning.md]: Yeah.

Comment (alodahl) [reply roles-in-tech/machine-learning.md]: If this is appearing, you probably don’t have the right file...

File: projects/eventonica-project.md (hunk 5 above)
Comment (alodahl) [reply projects/eventonica-project.md]: This is the wrong version; put your changes in the master version, please.

File: recursion/recursion.md [hunk 10]
```diff
@@ -7,11 +7,10 @@ About 4.5 hours
...
 ### Prerequisites
-- JS 1 lesson
-- JS 2 lesson
-- JS 3 lesson
-- JS 4 lesson
-- Score of __15__ or higher at Coding Challege Assesment, Level 1
+- [JavaScript lesson 1](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md)
+- [JavaScript lesson 2](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-2.md)
+- [JavaScript lesson 3](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-3.md)
+- [JavaScript lesson 4](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-4.md)
```
Comment (alodahl) [inline recursion/recursion.md]: 🙌🏻

File: roles-in-tech/machine-learning.md [hunk 11]
```diff
@@ -36,10 +39,35 @@
...
+
```
Comment (alodahl) [inline roles-in-tech/machine-learning.md]: Forget removing the link; please put your prerequisites in the master version of the file. This is the wrong version.

**Comment by rudri182** [comment]
I have created new PR for the issue of eventonica-project.md . Please review it. PR number is #775.

New PR generated for machine-learning.md. Please review it. PR is #776.

File: recursion/recursion.md (hunk 10 above)
Comment (rudri182) [reply recursion/recursion.md]: 😃

**Comment by alodahl** [comment]
@rudri182 please no additional PRs!  What I meant was, copy the code from the master branch:
https://github.com/Techtonica/curriculum/blob/master/projects/eventonica-project.md
Click the "raw" button, copy the code, and use it to replace your eventonica-project.md file. Add it to your Linking-in-prerequisite-sections branch, and commit so that you're making changes to the *up-to-date* version of the file. Do the same for the machine-learning.md.

**Comment by rudri182** [comment]
I have modified both the outlines as per the comments. Please review it and suggest if there are any changes required. Hope that suggested changes are done properly. 😊 

[evetonica-project](https://github.com/Techtonica/curriculum/blob/b5a3342dedd9a700fc728f502bd7b562e1a25818/projects/eventonica-project.md)

[machine-learning.md](https://github.com/Techtonica/curriculum/blob/b395341effd9dc326e78a72173142488c970cd12/roles-in-tech/machine-learning.md)

**Comment by rudri182** [comment]
I have closed those PRs that were invalid for this issue.

**Comment by rudri182** [comment]
I think I have resolved conflicts that were suggested by @saberprashant. 🤔  

From next time I will take care of white spaces. Thanks for guiding me and waiting for changes I do and push it.🎊

**Comment by Kundan28** [comment]
@rudri182 There is 1 conflict left. So please resolve that too.

**Comment by rudri182** [comment]
The conflicts are resolved in machine-learning.md.

File: roles-in-tech/machine-learning.md [hunk 12]
```diff
@@ -9,7 +9,8 @@
 
 ### Prerequisites
 
-- The main prerequisite for machine learning is data analysis.
+- The main prerequisite for machine learning is [data analysis](https://github.com/Techtonica/curriculum/blob/master/roles-in-tech/data-science.md).
```
Comment (alodahl) [inline roles-in-tech/machine-learning.md]: 🎯

**Comment by saberprashant** [comment]
Hi @rudri182 , really great work on this one 😄  🎉 . I am sorry that I took a lot of time to approve the changes you made. But, I am glad that you changed everything which I suggested, so yeah congrats and keep it up.
Cheers 🍺 🎉
